plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/MavenHostedFacet.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.repository.Facet;
+
+/**
+ * Maven hosted facet, present on all Maven hosted-type repositories.
+ *
+ * @since 3.0
+ */
+@Facet.Exposed
+public interface MavenHostedFacet
+    extends Facet
+{
+  /**
+   * Rebuilds/updates Maven metadata.
+   *
+   * @param update      if {@code true}, updates existing metadata, otherwise overwrites them with newly generated
+   *                    ones.
+   * @param groupId     scope the work to given groupId.
+   * @param artifactId  scope the work to given artifactId (groupId must be given).
+   * @param baseVersion scope the work to given baseVersion (groupId and artifactId must ge given).
+   */
+  void rebuildMetadata(boolean update,
```

Comment:
I think `update` should not be expose to client, but instead
`update = Strings.isEmptyOrNull(groupId) && Strings.isEmptyOrNull(artifactId) && Strings.isEmptyOrNull(version)`

As in a moment you "limit" the scope of rebuild, you MUST update (due to reasons above). And when you do it "for whole repo" (GAV all nulls), then you MUST create consistent md for whole repo == ie. drop refs to non existent plugins.